File: CS/InstantPrintReportsV2Example.Module/Controllers/InstantPrintReportController.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.ReportsV2;
using DevExpress.ExpressApp.Actions;
using DevExpress.Persistent.Base;
using InstantPrintReportsV2Example.Module.BusinessObjects;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace InstantPrintReportsV2Example.Module.Controllers {
    public abstract class PrintContactsController : ObjectViewController<ListView, Contact> {
        public PrintContactsController() {
            SimpleAction printAction = new SimpleAction(this, "PrintContacts", PredefinedCategory.Reports);
            printAction.ImageName = "Action_Printing_Print";
            printAction.Execute += delegate(object sender, SimpleActionExecuteEventArgs e) {
                IObjectSpace objectSpace = ReportDataProvider.ReportObjectSpaceProvider.CreateObjectSpace(typeof(ReportDataV2));
                IReportDataV2 reportData = objectSpace.FindObject<ReportDataV2>(
                    new BinaryOperator("DisplayName","Contacts Report"));
                if (reportData == null) {
                    throw new UserFriendlyException("Cannot find the 'Contacts Report' report.");
                }
                else {
                    PrintReport(reportData);
                }
            };
        }
        protected abstract void PrintReport(IReportDataV2 reportData);
    }
}

File: VB/InstantPrintReportsV2Example.Module/Controllers/InstantPrintReportController.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.ReportsV2
Imports DevExpress.ExpressApp.Actions
Imports DevExpress.Persistent.Base
Imports InstantPrintReportsV2Example.Module.BusinessObjects
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.BaseImpl

Namespace InstantPrintReportsV2Example.Module.Controllers
    Public MustInherit Class PrintContactsController
        Inherits ObjectViewController(Of ListView, Contact)

        Public Sub New()
            Dim printAction As New SimpleAction(Me, "PrintContacts", PredefinedCategory.Reports)
            printAction.ImageName = "Action_Printing_Print"
            AddHandler printAction.Execute, Sub(sender As Object, e As SimpleActionExecuteEventArgs)
                Dim objectSpace As IObjectSpace = ReportDataProvider.ReportObjectSpaceProvider.CreateObjectSpace(GetType(ReportDataV2))
                Dim reportData As IReportDataV2 = objectSpace.FindObject(Of ReportDataV2)(New BinaryOperator("DisplayName","Contacts Report"))
                If reportData Is Nothing Then
                    Throw New UserFriendlyException("Cannot find the 'Contacts Report' report.")
                Else
                    PrintReport(reportData)
                End If
            End Sub
        End Sub
        Protected MustOverride Sub PrintReport(ByVal reportData As IReportDataV2)
    End Class
End Namespace
